### PR TITLE
Update Daml Shell docs

### PR DESCRIPTION
**Pull request by iggy-da**
Updates the docs with the latest Daml Shell features, including new asciinema gifs.

I've updated the `2.9.0` docs as well as the `3.1` docs with the same content. Please let me know if that is the correct approach.

### CODE REVIEW

**Comment by yves-da** [comment]
Yes, that's the right approach as long as there are dedicated documentation version directories in the `main` branch.

**Comment by yves-da** [comment]
Please take note of
* #730

Depending on how long it takes to review this PR, you'll also have to add your changes to the new 2.10.0 folder. Thanks.

File: docs/3.1/docs/tools/daml-shell/index.rst [hunk 1]
```diff
@@ -97,17 +98,25 @@ the screen suggests possible command options.
...
+       can be used as filters and will receive the command output as stdin
 
 Filters
 =======
 
 You can pipe command output through one or more ``filters`` using the
 ``|`` (pipe) symbol. For example,
-``active Foo:Bar | csv | export ~/my_foos.csv``.
+``active Foo:Bar | csv | export ~/my_foos.csv``. If the filter is not
+recognized as a built-in filter, Daml Shell will use the system binary
```
Comment (carrielaben-da) [inline docs/3.1/docs/tools/daml-shell/index.rst:118]:
```suggestion
recognized as a built-in filter, Daml Shell uses the system binary
```

File: docs/3.1/docs/tools/daml-shell/index.rst [hunk 2]
```diff
@@ -97,17 +98,25 @@ the screen suggests possible command options.
...
+of the same name as the filter if it exists.
```
Comment (carrielaben-da) [inline docs/3.1/docs/tools/daml-shell/index.rst:119]:
```suggestion
of the same name as the filter if such a binary exists.
```

File: docs/2.9.0/docs/tools/daml-shell/index.rst [hunk 3]
```diff
@@ -97,17 +98,25 @@ the screen suggests possible command options.
...
+       can be used as filters and will receive the command output as stdin
 
 Filters
 =======
 
 You can pipe command output through one or more ``filters`` using the
 ``|`` (pipe) symbol. For example,
-``active Foo:Bar | csv | export ~/my_foos.csv``.
+``active Foo:Bar | csv | export ~/my_foos.csv``. If the filter is not
+recognized as a built-in filter, Daml Shell will use the system binary
```
Comment (carrielaben-da) [inline docs/2.9.0/docs/tools/daml-shell/index.rst:118]:
```suggestion
recognized as a built-in filter, Daml Shell uses the system binary
```

File: docs/2.9.0/docs/tools/daml-shell/index.rst [hunk 4]
```diff
@@ -97,17 +98,25 @@ the screen suggests possible command options.
...
+of the same name as the filter if it exists.
```
Comment (carrielaben-da) [inline docs/2.9.0/docs/tools/daml-shell/index.rst:119]:
```suggestion
of the same name as the filter if such a binary exists.
```

**Review by carrielaben-da — APPROVED** [review]
Two small changes, then GTG.

**Pull request merged**